src/components/WalletAssets.vue

```diff
@@ -199,13 +209,38 @@ export default class WalletAssets extends Mixins(LoadingMixin, FormattedAmountMi
     this.navigate({ name: RouteNames.AddAsset });
   }
 
-  updateAssets(): void {
-    const assetsAddresses = this.assetList.map((asset) => asset.address);
+  showAsset(asset: AccountAsset) {
+    // filter
+    const tokenType = this.filters.option;
+    const showWhitelistedOnly = this.filters.verifiedOnly;
+    const hideZeroBalance = this.filters.zeroBalance;
 
-    if (assetsAddresses.length) {
-      api.assets.accountAssetsAddresses = assetsAddresses;
-      api.assets.updateAccountAssets();
+    // asset
+    const isNft = api.assets.isNft(asset);
+    const isWhitelisted = api.assets.isWhitelist(asset, this.whitelist);
+    const hasZeroBalance = asset.balance.total === '0';
+
+    if (tokenType === WalletFilteringOptions.TOKEN && isNft) {
+      return false;
+    }
+
+    if (tokenType === WalletFilteringOptions.NFT && !isNft) {
+      return false;
+    }
+
+    if (!isWhitelisted && showWhitelistedOnly) {
+      return false;
     }
+
+    if (hideZeroBalance && hasZeroBalance) {
+      return false;
+    }
+
+    return true;
+  }
+
+  updated(): void {
+    this.numberOfRenderedAssets = this.walletAssets.length;
```

Comment:
Do we need it?

Reply:
Based on that we are figuring out whether to show 'No tokens found' or not

Reply:
not relevant